### ISSUE TITLE
Add handling for channels in different workspaces that share the same ID

### ISSUE THREAD

**Issue by mvlasovatl** [atlassian-labs/atlassian-slack-integration-server#145]
Slack supports the concept of shared channels, when the same channel is available in multiple workspaces. Jira Slack plugin fails if such set up is configured. These are the stacktraces that may be noticed in logs in such cases:
```
 java.lang.IllegalStateException: Duplicate key Conversation(id=GK9DTHU5V, ...)
 	at java.util.stream.Collectors.lambda$throwingMerger$0(Collectors.java:133)
 	at java.util.HashMap.merge(HashMap.java:1256)
 	at java.util.stream.Collectors.lambda$toMap$58(Collectors.java:1320)
 	at java.util.stream.ReduceOps$3ReducingSink.accept(ReduceOps.java:169)
 	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
 	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
 	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
 	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1384)
 	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:482)
 	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
 	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:708)
 	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
 	at java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:499)
 	at com.atlassian.plugins.slack.api.client.ConversationLoaderHelper.conversationsAndLinksById(ConversationLoaderHelper.java:112)
 	at com.atlassian.jira.plugins.slack.manager.impl.DefaultProjectConfigurationManager.toProjectConfigurationDTOs(DefaultProjectConfigurationManager.java:340)
 	at com.atlassian.jira.plugins.slack.manager.impl.DefaultProjectConfigurationManager.getConfigurations(DefaultProjectConfigurationManager.java:100)
 	at com.atlassian.jira.plugins.slack.web.contextproviders.SlackGlobalAdminDataProvider.getContextMap(SlackGlobalAdminDataProvider.java:57)
```
and another one
```
An error occurred rendering com.atlassian.jira.plugins.jira-slack-server-integration-plugin:slack-space-to-channel-global. Ignoring
java.lang.IllegalStateException: Duplicate key C023DDE3M7W (attempted merging values [com.atlassian.jira.plugins.slack.model.dto.ConfigurationGroupDTO@71e2e75a] and [com.atlassian.jira.plugins.slack.model.dto.ConfigurationGroupDTO@62206cf8])
	at java.base/java.util.stream.Collectors.duplicateKeyException(Unknown Source)
	at java.base/java.util.stream.Collectors.lambda$uniqKeysMapAccumulator$1(Unknown Source)
	at java.base/java.util.stream.ReduceOps$3ReducingSink.accept(Unknown Source)
	at java.base/java.util.Iterator.forEachRemaining(Unknown Source)
	at java.base/java.util.Spliterators$IteratorSpliterator.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
	at com.atlassian.jira.plugins.slack.model.dto.ProjectToChannelConfigurationDTO.<init>(ProjectToChannelConfigurationDTO.java:39)
	at com.atlassian.jira.plugins.slack.manager.impl.DefaultProjectConfigurationManager.toProjectToChannelConfigurationDTO(DefaultProjectConfigurationManager.java:154)
	at com.atlassian.jira.plugins.slack.manager.impl.DefaultProjectConfigurationManager.getConfigurations(DefaultProjectConfigurationManager.java:116)
```
As a temporary workaround user may delete duplicating records from the database manually, but as a long-term solution the app logic should be changed to handle this case without errors.

**Comment by mvlasovatl** [comment]
This issue was tackled in #153 and #164 that were merged to `dev`. It is pending next release now.

**Comment by mvlasovatl** [comment]
The fix was included into Bitbucket Server for Slack v.4.0.1.

**Comment by mvlasovatl** [comment]
The fix was included into Jira Server for Slack 2.0.23 and Confluence Server for Slack 2.1.4
